recipe/recipe.yaml

```diff
@@ -171,6 +171,7 @@ outputs:
         - jax >=0.7.0
         - lmfit >=1.3.4,<1.4
         - lsstdesc.coord >=1.3.0
+        - lsst-documenteer-guide >=2
```

Comment:
I think this needs to be 2.2?  Also `documenteer` should be removed, and the `sphinx-doxylink` thing added.